app/models/garden.rb

```diff
@@ -4,6 +4,8 @@ class Garden
   belongs_to :user
   validates_presence_of :user
 
+  after_save :reindex_guides
```

Comment:
That's something I need to look into.  I know with resque or sidekiq you can set it up to only run one of these jobs at a time.  I'm not sure about delayed job.